core/amazon/index.js

```diff
@@ -165,6 +174,11 @@ var amazonProduct = {
                                             && item.ItemAttributes[0].Title && item.ItemAttributes[0].Title[0]
                                         }
                                     } else {
+                                        // Check if variation values are too long or too numerous for selection through conversation
+                                        if (value.length > 20 || Object.keys(ref).length == 9) {
```

Comment:
Why Object.keys(ref).length == 9? I think it should be Object.keys(ref).length > 9. If it only has 9 variations to select, it should still work right?